src/bentoml/_internal/cloud/yatai.py

```diff
@@ -873,19 +873,21 @@ def _do_pull_model(
         context: str | None = None,
         query: str | None = None,
     ) -> Model:
+        _tag = Tag.from_taglike(tag)
         try:
-            model = model_store.get(tag)
-            if not force:
-                self.log_progress.add_task(
-                    f'[bold blue]Model "{tag}" already exists locally, skipping'
-                )
-                return model
-            else:
-                model_store.delete(tag)
+            model = model_store.get(_tag)
         except NotFound:
-            pass
+            model = None
+        else:
+            if _tag.version not in (None, "latest"):
+                if not force:
+                    self.log_progress.add_task(
+                        f'[bold blue]Model "{tag}" already exists locally, skipping'
+                    )
+                    return model
+                else:
+                    model_store.delete(tag)
```

Comment:
You will need to also add this to `_internal/cloud/bentocloud.py` as well 😄

Reply:
Oh yes, may be we could make it DRYer in the future.

Reply:
I think the idea for yatai.py and bentocloud.py is to have different implementation, but currently the bentocloud push for both model and bento are the same with yatai